local/src/core/ProxyVerifier.cc

```diff
@@ -1420,23 +1551,52 @@ HttpFields::parse_fields_and_rules(YAML::Node const &fields_rules_node,
                    node.Mark());
       continue;
     }
+
     TextView name{
         HttpHeader::localize_lower(node[YAML_RULE_NAME_KEY].Scalar())};
-    TextView value{HttpHeader::localize(node[YAML_RULE_DATA_KEY].Scalar())};
-    _fields.emplace(name, value);
-    if (node_size == 2 && assume_equality_rule) {
-      _rules.emplace(name, RuleCheck::make_equality(name, value));
-    } else if (node_size == 3) {
-      // Contans a verification rule.
-      TextView rule_type{node[YAML_RULE_TYPE_KEY].Scalar()};
-      std::shared_ptr<RuleCheck> tester =
-          RuleCheck::find(name, value, rule_type);
-      if (!tester) {
-        errata.error("Field rule at {} does not have a valid flag ({})",
-                     node.Mark(), rule_type);
-        continue;
-      } else {
-        _rules[name] = tester;
+    const YAML::Node ValueNode{node[YAML_RULE_DATA_KEY]};
+    if (ValueNode.IsScalar()) {
+      // There's only a single value associated with this field name.
+      TextView value{HttpHeader::localize(node[YAML_RULE_DATA_KEY].Scalar())};
+      _fields.emplace(name, value);
+      if (node_size == 2 && assume_equality_rule) {
+        _rules.emplace(name, RuleCheck::make_equality(name, value));
+      } else if (node_size == 3) {
+        // Contans a verification rule.
+        TextView rule_type{node[YAML_RULE_TYPE_KEY].Scalar()};
+        std::shared_ptr<RuleCheck> tester =
+            RuleCheck::make_rule_check(name, value, rule_type);
+        if (!tester) {
+          errata.error("Field rule at {} does not have a valid flag ({})",
+                       node.Mark(), rule_type);
+          continue;
+        } else {
+          _rules.emplace(name, tester);
+        }
+      }
+    } else if (ValueNode.IsSequence()) {
+      // There's a list of values associated with this field. This
+      // indicates duplicate fields for the same field name.
+      std::list<TextView> values;
+      for (auto const &value: ValueNode) {
+        TextView localized_value{HttpHeader::localize(value.Scalar())};
+        values.emplace_back(localized_value);
+        _fields.emplace(name, localized_value);
+      }
+      if (node_size == 2 && assume_equality_rule) {
+        _rules.emplace(name, RuleCheck::make_equality(name, values));
```

Comment:
You could consider storing the list in the field object, and making the `RuleCheck` depend on a reference or pointer to that.

Reply:
Or move the `list` in to the equality checker.